### DISCUSSION TITLE
Vendor class with ISO endpoints

### DISCUSSION THREAD

**Discussion by MikeEmbedded** [hathach/tinyusb#2678] [Q&A]
Hi.

I successfully run Vendor class device on USB HS using STM32H723 + USB3320.

Vendor class has BULK endpoints and it works fine.

But if I change descriptor and some code to have Vendor class with ISO endpoints then it enumerates successfully but there is no any data received(tud_vendor_rx_cb is not called when I send some data from PC).

So, is there any example or any possibility to use Vendor device class with ISO endpoints?

Thanks!

**Comment by hathach** [comment]
This is not common, you should implement your own driver in application using callback, there is a note for that in readme.
